common/src/IO/WalTextureReader.cpp

```diff
@@ -81,20 +81,24 @@ namespace TrenchBroom {
             const auto width = reader.readSize<uint32_t>();
             const auto height = reader.readSize<uint32_t>();
 
-            if (!m_palette.initialized()) {
-                return nullptr;
-            }
-
             const auto mipLevels = readMipOffsets(MaxMipLevels, offsets, width, height, reader);
             Assets::setMipBufferSize(buffers, mipLevels, width, height, GL_RGBA);
 
             reader.seekForward(32 + 2 * sizeof(uint32_t)); // animation name, flags, contents
             assert(reader.canRead(3 * 256));
 
-            const auto palette = Assets::Palette::fromRaw(3 * 256, reader.cur<unsigned char>());
-            readMips(palette, mipLevels, offsets, width, height, reader, buffers, averageColor);
+            if (m_palette.initialized()) {
```

Comment:
The old behaviour should be kept (prefer the embedded palette) so that all episodes display correctly in the texture browser. Is the issue that some skins for .dkm models lack an embedded palette?

Reply:
I'm not sure about preferring the embedded palette for textures. I can't find the comment, but I remember it was explained to me that the palette is episode specific, and is found using a worldspawn key which is supported by TB.

Reply:
This is the Dkt .wal info I guess?
https://gist.github.com/DanielGibson/a53c74b10ddd0a1f3d6ab42909d5b7e1

I prefer the embedded palette because it works and is tested (we've been doing that since before the Daikatana PR was merged), it keeps things simpler in TB, and lets the texture browser show everything correctly regardless of the map.

If we change it to prefer the worldspawn palette, all textures would need to reload when the worldspawn key is edited. We would also need to include the logic mentioned in the gist above that models always use the embedded palette, but map textures use the worldspawn key one.

Reply:
But if the game uses the global palette, then should we not do that as well? Otherwise the user may be in for bad surprises.

Reply:
Right, after a bit of back and forth on this, I have decided to remove any special handling for Daikatana palettes and just use the embedded palettes instead, which works perfectly fine.